### PR TITLE
Add LBSubnetPath under VPC status

### PR DESCRIPTION
**Pull request by lxiaopei**
For AVI integration, we need to set Avi Subnet Path under VPC CR Status.

### CODE REVIEW

File: pkg/apis/v1alpha1/vpc_types.go [hunk 1]
```diff
@@ -39,8 +39,10 @@ type VPCStatus struct {
 	// NSX VPC Policy API resource path.
 	NSXResourcePath string `json:"nsxResourcePath"`
 	// Default SNAT IP for private Subnets
-	DefaultSNATIP string            `json:"defaultSNATIP"`
-	CIDRsUsage    VPCCIDRsUsageInfo `json:"cidrsUsage"`
+	DefaultSNATIP string `json:"defaultSNATIP"`
+	// NSX PolicyPath for AVI private Subnet
+	LBSubnetPath string            `json:"lbSubnetPath"`
+	CIDRsUsage   VPCCIDRsUsageInfo `json:"cidrsUsage"`
```
Comment (jianjuns) [inline pkg/apis/v1alpha1/vpc_types.go:45]: In our offline discussion, we want to move this to another CRD?

File: pkg/apis/v1alpha1/vpc_types.go [hunk 2]
```diff
@@ -39,8 +39,10 @@ type VPCStatus struct {
 	// NSX VPC Policy API resource path.
 	NSXResourcePath string `json:"nsxResourcePath"`
 	// Default SNAT IP for private Subnets
-	DefaultSNATIP string            `json:"defaultSNATIP"`
-	CIDRsUsage    VPCCIDRsUsageInfo `json:"cidrsUsage"`
+	DefaultSNATIP string `json:"defaultSNATIP"`
+	// NSX PolicyPath for AVI private Subnet
```
Comment (jianjuns) [inline pkg/apis/v1alpha1/vpc_types.go:43]: AVI -> LB

File: pkg/apis/v1alpha1/vpc_types.go [hunk 3]
```diff
@@ -39,8 +39,10 @@ type VPCStatus struct {
 	// NSX VPC Policy API resource path.
 	NSXResourcePath string `json:"nsxResourcePath"`
 	// Default SNAT IP for private Subnets
```
Comment (jianjuns) [inline pkg/apis/v1alpha1/vpc_types.go:41]: Add "." to the end of the sentence.

File: pkg/apis/v1alpha1/vpc_types.go [hunk 4]
```diff
@@ -39,8 +39,10 @@ type VPCStatus struct {
...
-	DefaultSNATIP string            `json:"defaultSNATIP"`
-	CIDRsUsage    VPCCIDRsUsageInfo `json:"cidrsUsage"`
+	DefaultSNATIP string `json:"defaultSNATIP"`
+	// NSX PolicyPath for AVI private Subnet
+	LBSubnetPath string            `json:"lbSubnetPath"`
```
Comment (jianjuns) [inline pkg/apis/v1alpha1/vpc_types.go:44]: Could we add the LBSubnetCIDR too? I assume there are cases we expect customers to manually create SecurityPolicy with the LB CIDR to allow traffic from LB to workloads?

Comment (lxiaopei) [reply pkg/apis/v1alpha1/vpc_types.go:44]: OK, add LBSubnetCIDR.

File: pkg/apis/v1alpha1/vpc_types.go (hunk 2 above)
Comment (lxiaopei) [reply pkg/apis/v1alpha1/vpc_types.go]: changed. Also remove private.

File: pkg/apis/v1alpha1/vpc_types.go (hunk 3 above)
Comment (lxiaopei) [reply pkg/apis/v1alpha1/vpc_types.go]: done, thanks @jianjuns

File: pkg/apis/v1alpha1/vpc_types.go (hunk 1 above)
Comment (lxiaopei) [reply pkg/apis/v1alpha1/vpc_types.go]: removed CIDRsUsage in VPC CR.

File: pkg/apis/v1alpha1/vpc_types.go [hunk 5]
```diff
@@ -38,23 +38,12 @@ type VPCStatus struct {
...
-type CIDRsUsageInfo struct {
-	// Allocated IPs in the IP Block (including IPs used by other VPCs).
-	Allocated int `json:"allocated"`
-	// Used IP for the VPC in the IP Block (IPs used only in current VPC).
-	Used int `json:"used"`
-	// Total IP number in the IP Block.
-	Total int `json:"total"`
+	// Default SNAT IP for private Subnets.
+	DefaultSNATIP string `json:"defaultSNATIP"`
+	// NSX PolicyPath for LB Subnet.
```
Comment (jianjuns) [inline pkg/apis/v1alpha1/vpc_types.go:43]: Let us confirm with AVI team they really need NSX Operator to create the Subnet and need the path or not.

File: pkg/apis/v1alpha1/vpc_types.go [hunk 6]
```diff
@@ -38,23 +38,12 @@ type VPCStatus struct {
...
+	LBSubnetPath string `json:"lbSubnetPath"`
+	// CIDR for LB Subnet.
```
Comment (jianjuns) [inline pkg/apis/v1alpha1/vpc_types.go:45]: LB Subnet -> the load balancer Subnet

Comment (lxiaopei) [reply pkg/apis/v1alpha1/vpc_types.go:45]: changed. thanks.

**Review by jianjuns — APPROVED** [review]

**Pull request merged**